agent/lm_agent/reconciliation.py

```diff
@@ -89,11 +96,52 @@ async def clean_booked_grace_time():
         # if the running_time is greater than the greatest grace_time, delete the booking for it
         if job["run_time_in_seconds"] > greatest_grace_time and greatest_grace_time != -1:
             await remove_booked_for_job_id(job_id)
+    cluster_name = await get_cluster_name()
+    await clean_bookings(squeue_result, cluster_name)
+
+
+async def clean_bookings(squeue_result, cluster_name):
+    cluster_bookings = [booking.job_id for booking in await get_bookings_from_backend(cluster_name)]
+    jobs_not_running = [job["job_id"] for job in squeue_result if job["state"] != "RUNNING"]
+    all_jobs_squeue = [job["job_id"] for job in squeue_result]
+    delete_booking_call = []
+    for job_id in cluster_bookings:
+        if job_id in jobs_not_running:
+            delete_booking_call.append(remove_booked_for_job_id(job_id))
+        if job_id not in all_jobs_squeue:
+            delete_booking_call.append(remove_booked_for_job_id(job_id))
```

Comment:
Seems like we could combine the two if clauses:

```suggestion
        if job_id in jobs_not_running or job_id not in all_jobs_squeue:
            delete_booking_call.append(remove_booked_for_job_id(job_id))
```